src/libcore/ptr/mod.rs

```diff
@@ -269,10 +269,28 @@ pub const fn slice_from_raw_parts<T>(data: *const T, len: usize) -> *const [T] {
 ///
 /// [`slice_from_raw_parts`]: fn.slice_from_raw_parts.html
 /// [`from_raw_parts_mut`]: ../../std/slice/fn.from_raw_parts_mut.html
+///
+/// # Examples
+///
+/// ```rust
+/// use std::ptr;
+///
+/// let x = &mut [5, 6, 7];
+/// let raw_pointer = x.as_mut_ptr();
+/// let slice = ptr::slice_from_raw_parts_mut(raw_pointer, 3);
+///
+/// unsafe {
+///     (*slice)[2] = 99; // assign a value at an index in the slice
```

Comment:
Now you have 2 mutable references to the same slice, which is against Rust's rules. Here it is safe, since you own the other mut reference, but I'm not sure we want that in the std documentation. I'm not the right person to decide though, I'll let another reviewer decide.

Reply:
This is not guaranteed to be safe. Simply having two mutable references that point to the same value and are alive at the same time is UB. You could make the case that `x` is not really "live" at the same time as `slice` because it is not used after slice is created. AFAIK, this is not guaranteed.

Examples like this need to discuss the potential for aliasing violations.

Reply:
Hm, this feels like it would not be sound. `x` is still accessible and this is asserting uniqueness. Owning the other mutable reference, to my knowledge, is irrelevant.

cc @RalfJung @rust-lang/wg-unsafe-code-guidelines -- can we get an assessment here?

Reply:
A mutable reference is indeed created here for `IndexMut`, I think. However, `slice` is "derived from `x` through `raw_pointer`, so this is not unlike
```rust
let x = &mut [...];
let slice = &mut *x;
```
which also has "two aliasing mutable references", but one is derived from the other 
*and they are not used in interleaved ways*, so we are good.

The moment `x` gets used again, `slice` becomes invalid (its "reborrow ends", except there is no borrow checker checking that). This is confirmed by Miri which [only errors on the last line here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=df45a276a0ca7e244ed2045fe3322e2e), so there was no error before.

Reply:
So should the example stay or should I find another way to illustrate the mutability of `slice_from_raw_parts_mut` ?

Reply:
Ah, right, I forgot about the interleaved uses constraint being the real problem. Thanks!

I think this example is fine.